tests/test_storageclass_deletion.py

```diff
@@ -0,0 +1,221 @@
+from utility import utils, templating
+from utility.utils import run_cmd
+from ocsci.testlib import tier1, ManageTest
+
+from ocs import ocp
+from openshift.dynamic import DynamicClient
+import kubernetes
+
+
+k8s_client = kubernetes.config.new_client_from_config()
+dyn_client = DynamicClient(k8s_client)
+log = logging.getLogger(__name__)
+
+PVC_YAML = os.path.join(
+    "templates", "PersistentVolumeClaim_new.yaml"
+)
+
+SC_YAML = os.path.join(
+    "templates", "StorageClass.yaml"
+)
+
+TEMP_PVC_YAML_FILE = '/tmp/pvc_test.yaml'
+TEMP_POD_YAML_FILE = '/tmp/pod_test.yaml'
+TEMP_SC_YAML_FILE = '/tmp/sc_test.yaml'
+
+
+PVC = ocp.OCP(
+    kind='PersistentVolumeClaim', namespace=defaults.TEST_NAMESPACE
+)
+PV = ocp.OCP(
+    kind='PersistentVolume', namespace=defaults.TEST_NAMESPACE
+)
+SC = ocp.OCP(
+    kind='StorageClass', namespace=defaults.TEST_NAMESPACE
+)
+NAMESPACE = ocp.OCP(
+    kind='Project', namespace=defaults.TEST_NAMESPACE
+)
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Deleting a Storage class
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(teardown)
+    setup(self)
+
+
+def teardown(self):
+    """
+    Remove the resources after execution of tests
+    """
+    assert delete_storageclass(self.data['sc_name']), \
+        "Deletion of rbd storage class failed"
+    assert delete_namespace(self.data['project_name']), \
+        "Deletion of namespace failed"
+    utils.delete_file(TEMP_PVC_YAML_FILE)
+    utils.delete_file(TEMP_SC_YAML_FILE)
+
+
+def setup(self):
+    assert create_namespace(**self.data)
+    assert create_storageclass(**self.data)
+
+
+def create_namespace(**kwargs):
+    """
+    Creates a project if it is not already available
+    Args:
+        **kwargs:
+    Returns:
+        (bool): True namespace creation is successful,
+                False if the namespace creation is not successfull.
+    """
+    project_name = (kwargs['project_name'])
+    project_get = NAMESPACE.get()
+    namespaces = [item['metadata']['name'] for item in project_get['items']]
+    log.info(f'checking if project {project_name} already exists')
+    if project_name in namespaces:
+        log.info(
+            f'project {project_name} exists, using the existing namespace'
+        )
+        return True
+    else:
+        log.info(f'creating a new project {project_name}')
+        return run_cmd(f'oc new-project {project_name}')
+
+
+def delete_namespace(project_name):
+    """
+    Deletes the project
+    Args:
+        project_name (str): Project to be deleted
+    Returns:
+        (bool): True if deletion is successful, False otherwise
+    """
+
+    return NAMESPACE.delete(resource_name=project_name)
+
+
+def create_storageclass(**kwargs):
+    """
+    Creates a storage class
+    Args:
+        **kwargs:
+    Returns:
+        (bool): True if creation is successful, False otherwise
+    """
+
+    sc_name = (kwargs['sc_name'])
```

Comment:
As already discussed, no need to check if a resource already exists or not.

Reply:
done using helper.py